src/conn/conn_dhandle.c

```diff
@@ -17,7 +17,11 @@ __conn_dhandle_destroy(WT_SESSION_IMPL *session, WT_DATA_HANDLE *dhandle)
 {
 	WT_DECL_RET;
 
-	WT_WITH_DHANDLE(session, dhandle, ret = __wt_btree_discard(session));
+	if (WT_PREFIX_MATCH(dhandle->name, "file:"))
```

Comment:
It would be nice if this wasn't a string comparison, and there were an abstraction. Maybe a construct like:

```
WT_RET(__wt_get_handle_type(session, &type));
if (type == WT_DHANDLE_TYPE_BTREE)
    ....
else if (type == WT_DHANDLE_TYPE_TABLE)
    ....
```

Maybe with a default type of none? I guess it could even be a type field that is accessed directly?